<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"          default="OAK-D"           /> <!-- 'zed' or 'zedm' -->
    <arg name="tf_prefix"             default="oak"             />
    <arg name="base_frame"            default="oak-d_frame"     />
    <arg name="parent_frame"          default="oak-d-base-frame"/>
    
    <arg name="cam_pos_x"             default="0.0"             /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0"             /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0"             /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0"             /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0"             /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="0.0"             /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://depthai_examples/params/camera" />

    <include file="$(find depthai_bridge)/launch/urdf.launch">
        <arg name="camera_model"    value="$(arg  camera_model)" />
        <arg name="tf_prefix"       value="$(arg  tf_prefix)"  />
        <arg name="base_frame"      value="$(arg  base_frame)"   />
        <arg name="parent_frame"    value="$(arg  parent_frame)"/>
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"    />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"    />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"    />
        <arg name="cam_roll"        value="$(arg  cam_roll)"     />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"    />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"      />
    </include>


    <!-- Recom: 4K@30FPS@90%MJPEG ~ 40MB/s; 4K@24FPS@90%MJPEG ~ 30MB/s; 4K@15FPS@95%MJPEG ~ 30MB/s; 4K@15FPS@90%MJPEG ~ 16MB/s; 1080p@30FPS@95%MJPEG ~ 11.5MB/s; 1080p@30FPS@95%MJPEG ~ 6MB/s; -->

    <node name="oak_stereo_publisher" pkg="oak_driver" type="oak_stereo_node" output="screen" required="true">
        <param name="tf_prefix"        value="$(arg tf_prefix)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>
        <param name="device_name"       value="" />
        <param name="resolution"        value="720p" />
        <param name="fps"               value="30" />
        <param name="quality"           value="85" />
        <param name="codec"             value="MJPEG" />
        <param name="depth_aligned"     value="true" />
        <param name="out_LR"     	value="false" />
        <param name="rectify"     	value="true" />
        <param name="manual_focus"      value="125" />
        <param name="auto_white_balance_mode"      value="6" />
    </node>            
       

</launch>
